### PR TITLE
[WIP]Mix noise with audio

### PR DESCRIPTION
**Pull request by dataprowl**
This PR relates to #823 

**Included in this:**

- Simple utility to mix any background noise with audio at a specified SNR (in dB).

- Function to mix audio with white noise  at a specified SNR (in dB).

- Function to mix audio with red noise  at a specified SNR (in dB).

**Assumptions**

- Both background noise and audio are sampled at the sampling rate. 

- Both background noise and audio are mono channel.

### CODE REVIEW

**Comment by dataprowl** [comment]
> I think that the correct formule is snr = 10 ** (snr/10) instead 20.

@VictorBeraldo 

I agree! Regret for the mistake. Please see my change [here](https://github.com/pytorch/audio/pull/848/commits/e265ff3b4ec0a755463c38eeb0558133ea26219f#diff-292a2ef33b3ecf4c40cda1a4afe58760R2346). 

I hope this utility is similar to what you intend to integrate into torchaudio.

**Comment by vincentqb** [comment]
The addition of white noise is a simple operation, and wouldn't in itself warrant a new abstraction. That being said, it could make sense in the context of other noise types. Before going too deep in this pull request, we need to agree on a reference to look at and test against. Thoughts?

File: torchaudio/functional.py [hunk 1]
```diff
@@ -2314,3 +2321,93 @@ def vad(
...
+
+
+def add_background_noise(
+    waveform: Tensor,
+    noise: Tensor,
+    snr: float,
+    normalize: bool = True
+)-> Tensor:
+    r"""
+    Augment noise to waveform, with specified SNR value to generate degraded waveform.
```
Comment (vincentqb) [inline torchaudio/functional.py:2333]: Based on this description, it is not clear to me what this function would do beyond "addition". What use case do you have in mind for such a user-facing function? Would this be better as a private function?

Comment (vincentqb) [reply torchaudio/functional.py:2333]: Naively, this function would consume the color of the noise that a user wants to add?

**Comment by mthrok** [comment]
Closing as the work was initiated without specification. Please propose the specification in#823 first.

**Pull request closed without merging**